draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -1148,6 +1165,8 @@ Relying parties with additional sources of revocation such as {{CRLite}} or {{CR
 
 The transparency service does not prevent unauthorized certificates, but it aims to provide comparable security properties to Certificate Transparency {{?RFC6962}}. If a subscriber presents an acceptable Merkle Tree certificate to a relying party, the relying party should have assurance it was published in some form that monitors and, in particular, the subject of the certificate will be able to notice.
 
+One notable difference with Certificate Transparency is that the present transparency services do not publish the public keys, but rather hashes of them. This does not impede a relying party to detect a misissuance, by comparing with the hash of the expected public key. However, it does complicate studies (eg. {{SharedFactors}}) of weak public keys: researchers will have to retrieve the public keys from the TLS servers by connecting to them.
```

Comment:
Well, connecting to them may not work if the server is no longer presenting the keys. I also think it's worth a sentence for why we're sacrificing this. Perhaps...

```suggestion
One notable difference with Certificate Transparency is that transparency services do not publish the public keys, but rather hashes of them. This is intended to reduce serving costs, particularly with large post-quantum keys. Relying partings can still detect misissuance by looking for unexpected `subject_info_hash` values. However, this optimization complicates studies of weak public keys, e.g. {{SharedFactors}}. Such studies will have to retrieve the public keys separately, such as by connecting to the TLS servers, or fetching from the CA if it retains the unabridged assertion. This document does not define a mechanism for doing this.
```

I left open the possibility of the CA retaining it. Since the costs are more serving than storage, I think it'd be reasonable for a CA to retain it (possibly even expect them to, but we can leave that for later), and then maybe the researcher could work with a cooperating CA to fetch it. I'm assuming a one-off fetch is fine and it's really just persistently serving that much data to be costly.

Does that seem reasonable to you? We can also drop it, but I feel like if we just say "oh fetch it from the TLS server", we'll immediately invite all kinds of comments that the TLS server may not be serving it anymore. We're really saying we're okay not considering this use case to be in scope.